### PR TITLE
feat: e2e cypress testing

### PR DESCRIPTION
**Pull request by nameetjambla**
## Change Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist: (Check only applicable)

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: cypress.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
     "baseUrl": "http://localhost:4200",
     "experimentalStudio": true,
-    "testerUserName":"e2etestcarrier",
-    "testerPassword":"FleetHawks@1302"
+    "viewportHeight":1080,
+    "viewportWidth":1920
```
Comment (kunalfleethawks) [inline cypress.json]: Keep this as it is  and use it accordingly in all the test files
 "testerUserName":"e2etestcarrier",
     "testerPassword":"FleetHawks@1302"

File: cypress/integration/add-asset.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+it('add-asset credential', function() {
+   
+    /* ==== Generated with Cypress Studio ==== */
+    
+    cy.get(':nth-child(1) > .input-group > .form-control').clear();
+    cy.get(':nth-child(1) > .input-group > .form-control').type('hardeepcloud');
+    cy.get(':nth-child(2) > .input-group > .form-control').clear();
+    cy.get(':nth-child(2) > .input-group > .form-control').type('FleetHawks@1302');
+    cy.get('#btnsubmit').click();
+    cy.wait(5000);
```
Comment (kunalfleethawks) [inline cypress/integration/add-asset.js]: cy.get(':nth-child(1) > .input-group > .form-control').type(Cypress.config('testerUserName'));
cy.get(':nth-child(1) > .input-group > .form-control').type(Cypress.config('testerPassword'));

File: cypress/integration/add-driver.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
+describe('add-asset Test',function(){
+    beforeEach(function(){
+        cy.visit('http://cloudapp.fleethawks.com/#/Login/');
```
Comment (kunalfleethawks) [inline cypress/integration/add-driver.js]: cy.visit('#/Login/');
Change all the cy.visit() with actual pages.

File: cypress/integration/delete-asset.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+/* === Test Created with Cypress Studio === */
+it('delete asset', function() {
+  /* ==== Generated with Cypress Studio ==== */
+  cy.visit('http://cloudapp.fleethawks.com/#/Login/');
```
Comment (kunalfleethawks) [inline cypress/integration/delete-asset.js]: Change to relative url

File: cypress/integration/loginPassTest.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+/* === Test Created with Cypress Studio === */
+it('login test pass', function() {
+  /* ==== Generated with Cypress Studio ==== */
+  cy.visit('http://cloudapp.fleethawks.com/#/Login/');
```
Comment (kunalfleethawks) [inline cypress/integration/loginPassTest.js]: Change this to relative url

**Review by kunalfleethawks — CHANGES_REQUESTED** [review]
Please address the comments

File: cypress.json (hunk 1 above)
Comment (nameetjambla) [reply cypress.json]: Done

File: cypress/integration/add-asset.js (hunk 2 above)
Comment (nameetjambla) [reply cypress/integration/add-asset.js]: done

File: cypress/integration/add-driver.js (hunk 3 above)
Comment (nameetjambla) [reply cypress/integration/add-driver.js]: done

File: cypress/integration/delete-asset.js (hunk 4 above)
Comment (nameetjambla) [reply cypress/integration/delete-asset.js]: done

File: cypress/integration/loginPassTest.js (hunk 5 above)
Comment (nameetjambla) [reply cypress/integration/loginPassTest.js]: done

File: cypress/fixtures/example.json [hunk 6]
```diff
@@ -1,5 +0,0 @@
-{
-  "name": "Using fixtures to represent data",
```
Comment (kunalfleethawks) [inline cypress/fixtures/example.json:2]: Delete this file

File: cypress/fixtures/method.txt [hunk 7]
```diff
@@ -0,0 +1,152 @@
+import { browser, by, element, ElementFinder, ExpectedConditions, WebElementPromise } from 'protractor';
```
Comment (kunalfleethawks) [inline cypress/fixtures/method.txt:1]: delete this file

File: cypress/integration/Login Test/loginTests.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
+/* === Test Created with Cypress Studio === */
+it('login test', function () {
+    /* ==== Generated with Cypress Studio ==== */
+    cy.visit('https://devapp.fleethawks.com/#/Login');
+    cy.get(':nth-child(1) > .input-group > .form-control').clear();
```
Comment (kunalfleethawks) [inline cypress/integration/Login Test/loginTests.js]: describe (Login Tests):
it ( should allow user to login when valid credentails are provided
it( should fail with errormessage when invalid credentials are provided)
it(should give validation error message when either username or password is not provided

File: cypress/integration/Login Test/loginTests.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+it('login failed', function () {
+    cy.get(':nth-child(1) > .input-group > .form-control').clear();
+    cy.get(':nth-child(1) > .input-group > .form-control').type('kkkk');
+    cy.get(':nth-child(2) > .input-group > .form-control').clear();
+    cy.get(':nth-child(2) > .input-group > .form-control').type('1234');
+    cy.get('#btnsubmit').click();
+    cy.get('.text').contains('Incorrect username or password.');
+    /* ==== End Cypress Studio ==== */
+});
+it('login and then go to Map Dashdashboard and then add asset and then delete asset', function () {
```
Comment (kunalfleethawks) [inline cypress/integration/Login Test/loginTests.js]: Move this to New File
describe (“Asset tests”)
it should allow user to add asset
it should list the assed which was added (verify the added asset is listed by name/other properties)
it should allow user to search by asset name and other properties
it should allow user to delete asset
it should not be listed (verify assed which was added is not listed)
it should give validation error message when required fields are not provided

File: package.json [hunk 10]
```diff
@@ -80,7 +80,7 @@
   },
   "devDependencies": {
     "@angular-devkit/build-angular": "^0.803.29",
-    "@angular/cli": "^8.3.29",
+    "@angular/cli": "^11.2.7",
```
Comment (kunalfleethawks) [inline package.json]: revert this change

**Review by kunalfleethawks — CHANGES_REQUESTED** [review]

File: package.json [hunk 11]
```diff
@@ -97,14 +97,15 @@
     "codelyzer": "5.0.0",
     "concurrently": "^6.0.0",
     "cypress": "^6.8.0",
+    "cypress-file-upload": "^5.0.3",
     "jasmine-core": "3.4.0",
     "jasmine-spec-reporter": "4.2.1",
-    "karma": "4.1.0",
+    "karma": "^6.3.2",
```
Comment (kunalfleethawks) [inline package.json]: revert this

File: package.json [hunk 12]
```diff
@@ -97,14 +97,15 @@
...
     "karma-chrome-launcher": "2.2.0",
     "karma-coverage-istanbul-reporter": "2.0.1",
     "karma-jasmine": "2.0.1",
     "karma-jasmine-html-reporter": "1.4.0",
-    "protractor": "5.4.0",
+    "protractor": "^5.4.4",
```
Comment (kunalfleethawks) [inline package.json]: revert

File: report.20210405.141638.6720.0.001.json [hunk 13]
```diff
@@ -0,0 +1,424 @@
+
```
Comment (kunalfleethawks) [inline report.20210405.141638.6720.0.001.json]: delete this file

**Review by kunalfleethawks — CHANGES_REQUESTED** [review]
@nameetjambla please address the review comments

File: cypress/fixtures/example.json (hunk 6 above)
Comment (nameetjambla) [reply cypress/fixtures/example.json:2]: Issue Solved

File: package.json (hunk 10 above)
Comment (nameetjambla) [reply package.json]: Issue Solved

File: package.json (hunk 12 above)
Comment (nameetjambla) [reply package.json]: Issue Solved

File: report.20210405.141638.6720.0.001.json (hunk 13 above)
Comment (nameetjambla) [reply report.20210405.141638.6720.0.001.json]: Issue Solved

File: package.json (hunk 11 above)
Comment (nameetjambla) [reply package.json]: Issue Solved

File: cypress/fixtures/method.txt (hunk 7 above)
Comment (nameetjambla) [reply cypress/fixtures/method.txt:1]: Issue Solved

**Review by kunalfleethawks — APPROVED** [review]

**Pull request merged**